hwilib/hwwclient.py

```diff
@@ -1,82 +1,167 @@
+from typing import Dict, Optional, Union
+
 from .base58 import get_xpub_fingerprint_hex
 
-# This is an abstract class that defines all of the methods that each Hardware
-# wallet subclass must implement.
+
 class HardwareWalletClient(object):
+    """Create a client for a HID device that has already been opened.
+
+    This abstract class defines the methods
+    that hardware wallet subclasses should implement.
+    """
 
-    # device is an HID device that has already been opened.
-    def __init__(self, path, password, expert):
+    def __init__(self, path: str, password: str, expert: bool) -> None:
         self.path = path
         self.password = password
         self.message_magic = b"\x18Bitcoin Signed Message:\n"
         self.is_testnet = False
-        self.fingerprint = None
-        self.xpub_cache = {}
+        self.fingerprint: Optional[str] = None
+        # {bip32_path: xpub}
+        self.xpub_cache: Dict[str, str] = {}
         self.expert = expert
 
-    # Get the master BIP 44 pubkey
-    def get_master_xpub(self):
-        return self.get_pubkey_at_path('m/44\'/0\'/0\'')
-
-    # Get the master fingerprint
-    def get_master_fingerprint_hex(self):
-        master_xpub = self.get_pubkey_at_path('m/0h')['xpub']
+    def get_master_xpub(self) -> Dict[str, str]:
+        """Return the master xpub.
+
+        Return {"xpub": xpub}
+        """
+        # FIXME is "m/44'/0'/0'" just an example?
+        # how is testnet handled?
+        # shouldn't this be
+        # return self.get_pubkey_at_path(self.path)
+        # or according to get_master_fingerprint_hex
+        # return self.get_pubkey_at_path("m/0h")
+        return self.get_pubkey_at_path("m/44'/0'/0'")
```

Comment:
Re your questions: This follows the BIP 44 default, although it indeed should be different for testnet.